odoo/modules/module.py

```diff
@@ -401,19 +403,25 @@ def is_really_module(name):
             for mname in MANIFEST_NAMES:
                 if os.path.isfile(opj(dir, name, mname)):
                     return True
-        return [
-            clean(it)
+        return {
+            clean(it): os.sep.join([ad, it])
             for it in os.listdir(dir)
             if is_really_module(it)
-        ]
+        }
 
-    plist = []
+    modules = {}
     for ad in odoo.addons.__path__:
         if not os.path.exists(ad):
             _logger.warning("addons path does not exist: %s", ad)
             continue
-        plist.extend(listdir(ad))
-    return list(set(plist))
+        modules.update(listdir(ad))
```

Comment:
Nitpicking, we could simplify the `listdir` function very much, so much that we could get rid of it

```suggestion
        modules.update({
            module.removesuffix('.zip'): opj(ad, module)
            for module in os.listdir(ad)
            if any((os.path.isfile(opj(ad, module, mname))
                    for mname in MANIFEST_NAMES))
        })
```

Btw external question, does `is_really_module` works for zip files ?!

Reply:
I agree that this could be change, but this is outside of the scope of this pr, i'm trying to make minimal changes here in order to avoid being stuck because changing legacy code to deeply :) But I agree that this could benefit from some cleanup :)